### PR TITLE
Turn on several lint rules 🧹

### PR DESCRIPTION
**Pull request by ejizba**
Unlike the [last round](https://github.com/Azure/azure-functions-nodejs-worker/pull/437) of lint rules I added, none of these could be auto-fixed. However, they were still pretty trivial manual changes. I split each rule up into its own separate commit - I figured that might be easier to view.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,8 @@
     "devDependencies": {
         "@types/mocha": "^2.2.48",
         "@types/chai": "^4.2.22",
+        "@types/chai-as-promised": "^7.1.5",
+        "@types/mock-require": "^2.0.1",
```
Comment (hossam-nasr) [inline package.json:18]: Are the types packages required to satisfy the linting rules? If it's just for added elegance can we move this to a different PR?

File: src/Worker.ts [hunk 2]
```diff
@@ -39,7 +36,7 @@ export function startNodeWorker(args) {
         throw new InternalException(exception);
     }
 
-    const workerChannel = new WorkerChannel(workerId, eventStream, new FunctionLoader());
+    new WorkerChannel(workerId, eventStream, new FunctionLoader());
```
Comment (hossam-nasr) [inline src/Worker.ts:39]: If it's created and never used do we even need this line?

File: src/FunctionLoader.ts [hunk 3]
```diff
@@ -50,7 +49,7 @@ export class FunctionLoader implements IFunctionLoader {
         }
         const entryPoint = <string>(metadata && metadata.entryPoint);
         const userFunction = getEntryPoint(script, entryPoint);
-        if (!isFunction(userFunction)) {
+        if (typeof userFunction !== 'function') {
```
Comment (hossam-nasr) [inline src/FunctionLoader.ts:52]: Just out of curiosity, is there a difference between using `typeof userFunction !== 'function'` and `!(userFunction instanceof Function)`?

File: src/Worker.ts (hunk 2 above)
Comment (ejizba) [reply src/Worker.ts:39]: This is an anti-pattern that I would like to fix in a separate PR. The constructor for the WorkerChannel has side effects, even though the workerChannel is never used - which means it should be a function, not a class

File: package.json (hunk 1 above)
Comment (ejizba) [reply package.json:18]: They tell me if I'm importing the modules correctly which was very useful when fixing the "no-var-requires" rule: https://github.com/Azure/azure-functions-nodejs-worker/pull/535/commits/86cba78e9a42f216fa7f9ba097eb45c54071e180

I plan to rebase when merging this PR (unless you have major objections) and I feel like adding the types when turning on that rule makes sense. Each commit by itself is relatively small and encapsulated

File: src/FunctionLoader.ts (hunk 3 above)
Comment (ejizba) [reply src/FunctionLoader.ts:52]: Tbh, I've never seen anyone use `instanceof Function` 😝  I don't know all the nuances, but here are a few points of reference:
- The docs for "isFunction" say to use "typeof" https://nodejs.org/api/all.html#all_util_utilisfunctionobject
- https://stackoverflow.com/questions/899574/what-is-the-difference-between-typeof-and-instanceof-and-when-should-one-be-used

**Review by hossam-nasr — APPROVED** [review]
